### DISCUSSION TITLE
packetArrived not working

### DISCUSSION THREAD

**Discussion by KoYejune0302** [ANLAB-KAIST/KENSv3#196] [Project Q&A 2. Transport Layer (TCP)]
wireshark로 확인을 해보면 syn을 받은뒤 다시 syn, ack를 보내는게 찍혀있는데 코드에서 cout으로 출력을 해보면 syn을 받는 부분은 출력이 되는데 syn, ack를 받는 부분이 출력이 안되는 것으로 보아 sendPacket을 해도 packetArrived 호출이 되는것 같은데 어느 부분이 문제인지 모르겠습니다.

+
TCP_Log를 출력해보니 TCP_checksum에 문제가 있다는 것을 발견했는데, syn을 보낼 때와 동일한 방식으로 NetworkUtil::tcp_sum을 호출했는데 syn에서는 되고 syn, ack에서는 안되는 이유를 모르겠습니다.

<img width="740" alt="image" src="https://github.com/ANLAB-KAIST/KENSv3/assets/57734106/4d7c8b4d-05d6-443a-ab82-eaa66396576b">
<img width="1096" alt="image" src="https://github.com/ANLAB-KAIST/KENSv3/assets/57734106/2d5331a9-1653-41d4-b664-a08e74bffdb4">
<img width="597" alt="image" src="https://github.com/ANLAB-KAIST/KENSv3/assets/57734106/b2f4be86-bba5-4ea8-bf94-74dd9f06e127">

**Comment by Pusnow** [comment]
It seems your packets have wrong packet size (100). Please, check the value.
